tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/OrderItemDeleteSteps.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Data.SqlClient;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Requests;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Responses;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Steps.Common;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Support;
+using NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Utils;
+using NHSD.BuyingCatalogue.Ordering.Api.Testing.Data.Entities;
+using TechTalk.SpecFlow;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests.Steps
+{
+    [Binding]
+    internal sealed class OrderItemDeleteSteps
+    {
+        private readonly Request request;
+        private readonly Response response;
+        private readonly Settings settings;
+        private DeleteOrderItemRequest deleteOrderItemRequest;
+        private int originalOrderItemsCount;
+
+        public OrderItemDeleteSteps(
+            Request request,
+            Response response,
+            Settings settings)
+        {
+            this.request = request;
+            this.response = response;
+            this.settings = settings ?? throw new ArgumentNullException(nameof(settings));
+        }
+
+        [Given(@"the user creates a request to delete order item with catalogue item ID (.*) for order ID (\d{1,6})")]
+        public async Task GivenTheUserCreatesARequestToDeleteOrderItemForOrder(string catalogueItemId, int orderId)
+        {
+            try
+            {
+                originalOrderItemsCount = await OrderItemEntity.GetOrderItemCountForOrder(
+                        settings.ConnectionString,
+                        orderId);
+            }
+            catch (SqlException)
+            {
+            }
+
+            deleteOrderItemRequest = new DeleteOrderItemRequest(
+                request,
+                settings.OrderingApiBaseUrl,
+                catalogueItemId,
+                orderId);
+        }
+
+        [When(@"the user sends the delete order item request")]
+        public async Task WhenTheUserSendsTheDeleteOrderItemRequest()
+        {
+            await deleteOrderItemRequest.ExecuteAsync();
+        }
+
+        [Then(@"the expected order items and additional services are deleted")]
+        public async Task ThenTheExpectedOrderItemsAndAdditonalServicesAreDeleted()
```

Comment:
Spotted a typo on second glance, `Additional` is missing its last `i`.

Reply:
updated

Reply:
Is this one definitely updated? I can't see the change.